### PR TITLE
copying 0 element from an empty array should be a noop

### PR DESCRIPTION
**Pull request by fbertra**
Note: replace https://github.com/scala-native/scala-native/pull/268

### CODE REVIEW

File: unit-tests/src/main/scala/scala/ArrayCopySuite.scala [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+          arr(6) == 6 && arr(7) == 7 && arr(8) == 8 &&
+          arr(9) == 9
+    )
+  }
+
+  test("array[Int]: copy to another array") {
+    init(arr)
+    scala.Array.copy(arr, 0, arr2, 1, 10)
+    assert(
+        arr2(0) == 0 && arr2(1) == 0 && arr2(2) == 1 &&
```
Comment (pbatko) [inline unit-tests/src/main/scala/scala/ArrayCopySuite.scala]: these two `0`s next to each other, 
what about having sentinel values different from the other values?

File: unit-tests/src/main/scala/scala/ArrayCopySuite.scala [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+
+  test("array[Int]: from is null") {
+    assertThrows[java.lang.NullPointerException] {
+      scala.Array.copy(null, 0, arr2, 5, 2)
+    }
+  }
+
+  test("array[Int]: to is null") {
+    assertThrows[java.lang.NullPointerException] {
+      scala.Array.copy(arr, 0, null, 5, -2)
```
Comment (pbatko) [inline unit-tests/src/main/scala/scala/ArrayCopySuite.scala]: I think length should be positive  here

File: unit-tests/src/main/scala/scala/ArrayCopySuite.scala [hunk 3]
```diff
@@ -0,0 +1,435 @@
...
+
+  test("array[Int]: toPos + len > to.length") {
+    assertThrows[java.lang.IndexOutOfBoundsException] {
+      scala.Array.copy(arr, 0, arr2, 5, 10)
+    }
+  }
+
+  test("array[Int]: fromPos + len > from.length") {
+    assertThrows[java.lang.IndexOutOfBoundsException] {
+      scala.Array.copy(arr, 5, arr2, 5, 10)
```
Comment (pbatko) [inline unit-tests/src/main/scala/scala/ArrayCopySuite.scala]: I think `toPos + len` should be `< arr2.length`
similarly in `"array[Int]: fromPos is negative"` test case

**Comment by fbertra** [comment]
@pbatko, you're absolutely right, working on it.  thanks

**Comment by densh** [comment]
LGTM, thanks @fbertra !

**Comment by densh** [comment]
I think this PR officially gets the prize for the most issues closed by a simple fix. 🍰 👍

**Pull request merged**